project_resource_calendar/models/calendar_event.py

```diff
@@ -138,42 +150,39 @@ def _get_weekday(self):
 
     @api.multi
     @api.constrains('room_id', 'start', 'stop', 'equipment_ids')
-    def _check_room_id_double_book(self):
+    def _check_resources_double_book(self):
         for record in self:
-
-            if record._event_in_past() or record.state == 'draft':
+            if record._event_in_past() or record.state == 'cancelled':
                 continue
-
             room = record.room_id.filtered(
                 lambda s: s.allow_double_book is False
             )
             equipment = record.equipment_ids.filtered(
                 lambda s: s.allow_double_book is False
             )
-
             if not any(room) and not any(equipment):
                 continue
-
             overlaps = self.env['calendar.event'].search([
                 ('id', '!=', record.id),
                 ('start', '<', record.stop),
                 ('stop', '>', record.start),
             ])
-
             for resource in overlaps.mapped(lambda s: s.room_id):
-                raise ValidationError(
-                    _(
-                        'The room cannot be double-booked '
-                        'with any overlapping meetings or events.',
+                if resource.id == record.room_id.id:
+                    raise ValidationError(
+                        _(
+                            'The room, %s,  cannot be double-booked '
+                            'with any overlapping meetings or events.',
+                        ) % resource.name,
                     )
-                )
             for resource in overlaps.mapped(lambda s: s.equipment_ids):
-                raise ValidationError(
-                    _(
-                        'The resource cannot be double-booked '
-                        'with any overlapping meetings or events.',
+                if resource.id in record.equipment_ids.ids:
```

Comment:
Can we do this?
 if resource.id in record.equipment_ids.ids:?
I mean list.ids ?